### PR TITLE
2nd to 3rd person

### PR DESCRIPTION
**Pull request by Dangthrimble**
Replacing all references to the 2nd person with references to the 3rd person (excluding `authors = [ "Your name <you@example.com>" ]` and `file:///home/yourname/projects/hello_world` in `hello-cargo.md`). Have to confess there are a few instances that don't feel quite right so would appreciate your views on the endeavour!

r? @steveklabnik

### CODE REVIEW

File: src/doc/trpl/README.md [hunk 1]
```diff
@@ -34,11 +33,11 @@ is the first. After this:
...
-After reading this introduction, you’ll want to dive into either ‘Learn Rust’
-or ‘Syntax and Semantics’, depending on your preference: ‘Learn Rust’ if you
-want to dive in with a project, or ‘Syntax and Semantics’ if you prefer to
-start small, and learn a single concept thoroughly before moving onto the next.
-Copious cross-linking connects these parts together.
+After reading this introduction, we’ll want to dive into either ‘Learn Rust’ or
+‘Syntax and Semantics’, depending on our preference: ‘Learn Rust’ if we want to
+dive in with a project, or ‘Syntax and Semantics’ if we prefer to start small,
+and learn a single concept thoroughly before moving onto the next. Copious
+cross-linking connects these parts together.
```
Comment (steveklabnik) [inline src/doc/trpl/README.md]: So, while I'm generally super in favor of moving EVERYTHING to 'we', I think that this section should stay as-is. In this case, we are addressing the reader directly, they're the one making this choice. I think from here and above, 'you' is fine.

File: src/doc/trpl/README.md [hunk 2]
```diff
@@ -76,11 +75,11 @@ type inference to balance out the power of static typing with the verbosity of
...
 
 Rust prefers stack allocation to heap allocation: `x` is placed directly on the
-stack. However, the `Vec<T>` type allocates space for the elements of the
-vector on the heap. If you’re not familiar with this distinction, you can
-ignore it for now, or check out [‘The Stack and the Heap’][heap]. As a systems
-programming language, Rust gives you the ability to control how your memory is
-allocated, but when we’re getting started, it’s less of a big deal.
+stack. However, the `Vec<T>` type allocates space for the elements of the vector
+on the heap. If we’re not familiar with this distinction, we can ignore it for
+now, or check out [‘The Stack and the Heap’][heap]. As a systems programming
```
Comment (steveklabnik) [inline src/doc/trpl/README.md]: I would argue this is also directed at the reader

File: src/doc/trpl/README.md [hunk 3]
```diff
@@ -76,11 +75,11 @@ type inference to balance out the power of static typing with the verbosity of
...
+language, Rust gives us the ability to control how our memory is allocated, but
```
Comment (steveklabnik) [inline src/doc/trpl/README.md]: but this one is okay, for example

File: src/doc/trpl/README.md (hunk 2 above)
Comment (steveklabnik) [inline src/doc/trpl/README.md]: (the line above, that is, sorry, should have been on 79.)

File: src/doc/trpl/hello-world.md [hunk 4]
```diff
@@ -150,20 +151,19 @@ $ ./main  # or main.exe on Windows
...
 
-If you come from a dynamic language like Ruby, Python, or JavaScript,
-you may not be used to these two steps being separate. Rust is an
-‘ahead-of-time compiled language’, which means that you can compile a program,
-give it to someone else, and they don't need to have Rust installed. If you
-give someone a `.rb` or `.py` or `.js` file, they need to have a
-Ruby/Python/JavaScript implementation installed, but you just need one command
-to both compile and run your program. Everything is a tradeoff in language
-design, and Rust has made its choice.
+If we come from a dynamic language like Ruby, Python, or JavaScript, we may not
```
Comment (steveklabnik) [inline src/doc/trpl/hello-world.md]: This also feels more reader-directed to me. but the sentences that follow feel fine.

File: src/doc/trpl/hello-world.md [hunk 5]
```diff
@@ -150,20 +151,19 @@ $ ./main  # or main.exe on Windows
...
+language’, which means that we can compile a program, give it to someone else,
+and they don't need to have Rust installed. If we give someone a `.rb` or `.py`
+or `.js` file, they need to have a Ruby/Python/JavaScript implementation
+installed, but we just need one command to both compile and run our program.
+Everything is a tradeoff in language design, and Rust has made its choice.
 
-Congratulations! You have officially written a Rust program. That makes you a
-Rust programmer! Welcome. 🎊🎉👍
+Congratulations! We have officially written a Rust program. That makes us Rust
+programmers! Welcome. 🎊🎉👍
```
Comment (steveklabnik) [inline src/doc/trpl/hello-world.md]: same here, this should stay 'you'

File: src/doc/trpl/installing-rust.md [hunk 6]
```diff
@@ -24,9 +24,9 @@ $ sh rustup.sh
 
 [insecurity]: http://curlpipesh.tumblr.com
 
-If you're on Windows, please download the appropriate [installer][install-page].
+If we're on Windows, please download the appropriate [installer][install-page].
```
Comment (steveklabnik) [inline src/doc/trpl/installing-rust.md]: same

**Comment by steveklabnik** [comment]
This diff is awesome, thanks so much! Yeah, after reading over the literal changes, I'm with you: some feel strange. I think the ones that are specifically directed at a reader should stay 'you', imagine it's a conversation, just because you say 'we' for most things doesn't mean 'you' is always inappropriate.

Are there any that seemed off that I missed?

**Comment by steveklabnik** [comment]
Hey @Dangthrimble ! it doens't notify me when you push commits, so sorry that i missed this. I think this is good to go, can you please rebase and sqaush? r=me after :+1:

**Comment by Dangthrimble** [comment]
Rebase and squash? Don't understand!

Do I need to "r=" after every update? Do you mean "r?" or does "r=" specify who it is intended for whereas "r?" indicates a preference?

I'll get the hang of GitHub one of these days!

**Comment by steveklabnik** [comment]
Whoops, sorry! :)

"Rebase and squash" is a git term: http://blog.steveklabnik.com/posts/2012-11-08-how-to-squash-commits-in-a-github-pull-request

"r=me" is a Mozilla social convention that says, "Hey anyone else with commit privileges: I'm okay with this PR, but there's one or two tiny things that still need to be fixed. If the submitter fixes them before I get back, go ahead and approve this PR."

**Comment by steveklabnik** [comment]
@Dangthrimble i see you pushed some more commits, but they still need to be squashed. Let me know after you push things, I only get a notification on comments, not on new commits :/

**Comment by Dangthrimble** [comment]
Just trying to find some spare time!

**Comment by steveklabnik** [comment]
No worries! Just wanted to make sure I knew so we could get it in :)

**Comment by Dangthrimble** [comment]
Finally found some time to have a go at this but the git command line
is all gibberish to me! Till now I have done just about all by
"gitting"  using GitHub Desktop. Following the commands in your link,
how do I find out what to type in place of upstream and omgpull?
Thanks

----- Original Message -----

**Comment by Dangthrimble** [comment]
Just found an earlier email of yours that used "git remote -v" to
identify them. I get:
 > git remote -vorigin  https://github.com/Dangthrimble/rust.git
(fetch)origin  https://github.com/Dangthrimble/rust.git
(push)rust-lang       https://github.com/rust-lang/rust.git
(fetch)rust-lang       https://github.com/rust-lang/rust.git (push)
So I guess I would:
$ git fetch rust-lang$ git checkout origin $ git rebase
-i rust-lang/master$ git push origin origin -f
Does that look right?

----- Original Message -----

**Comment by steveklabnik** [comment]
That's right except you want to check out `master`. You check out the branch, not remote, and the branch of this PR is master:

2015-09-17-091903_189x45_scrot

**Comment by Dangthrimble** [comment]
I'm getting:

```
> git fetch rust-lang
> git checkout master
Already on 'master'
Your branch is ahead of 'origin/master' by 391 commits.
  (use "git push" to publish your local commits)
```

Those 391 commits worry me! Has something gone wrong or do I just go ahead?

**Comment by steveklabnik** [comment]
Well, that does make some sense, as we've put in a lot of development since this was opened :)

Regardless though, I've pulled in and rebased it myself: https://github.com/rust-lang/rust/compare/master...steveklabnik:Dangthrimble so don't worry about screwing something up, we can always fix it.

**Comment by steveklabnik** [comment]
(and if you get the chance, maybe jumping into IRC is easier than comments in this PR thread)

**Comment by steveklabnik** [comment]
@Dangthrimble would you like me to do this rebase myself so we can get this shipped, or are you going to have time soon?

**Comment by Dangthrimble** [comment]
I'm trying to help organise my daughter's wedding at the moment and I'm 225% loaded at work for the next few months, so spare time is proving elusive! Could you do it and I'll try and pick up sometime next year!!!

Thanks

**Comment by steveklabnik** [comment]
Oh wonderful! Congrats. :)

Thank you again for the patch, i'll make sure you're still the author in git.

**Comment by steveklabnik** [comment]
superceded by https://github.com/rust-lang/rust/pull/28930, thanks again!

**Comment by Ketsuban** [comment]
I'm unable to connect securely to Mozilla's IRC server for some reason so I'll leave my thoughts here, even though commenting on a committed pull request feels weird.

I'm not sure this was sufficiently thought through; I think the best way to show why is to provide examples and narrate. I apologise for the resulting chunkiness.

> This book will teach you about the Rust Programming Language.

Indeed, this is the goal of the book. Now let's look at the revised version of this sentence.

> This book will teach us about the Rust Programming Language.

If the author needs to be taught about Rust then they probably shouldn't be writing a book on it. (My understanding is the authorial voice is something akin to “the Rust community”, speaking from a position of knowledge.)

> Getting started - Set up our computer for Rust development.

Get your filthy hands off my computer! This was unobjectionable when it said “your”.

> If we're concerned about the potential insecurity of using `curl | sh`, please keep reading and see our disclaimer below.

If the author is concerned about the potential insecurity of using `curl | sh`, why are they doing it? The book should be addressing the reader's fears and concerns, not the author's own.

> If you're on Windows, please download the appropriate installer.

Notice this didn't get changed to “we”, changing person in mid-flow. It was consistent before, but now it jars. (Incidentally, microcomplaint about the pull request itself: “we” isn't third person, it's first person plural. Third person would be “they”.)

> If we decide we don't want Rust anymore, we'll be a bit sad, but that's okay.

Now you sound like you can't make up your mind.

**Comment by steveklabnik** [comment]
Thanks @Ketsuban !

I'm going to be going over all of this with the editors, so I'll mark your specific concerns, and we'll see. I think starting from consistency and then making exceptions makes for ultimately clear writing, but you're right that a few of these seem off.

**Comment by Ketsuban** [comment]
Alright, thanks. ☺

**Comment by KodrAus** [comment]
@steveklabnik I would think it would make more sense to start from consistency in the target audience for a particular statement, rather than consistency in wording for all statements.

**Pull request closed without merging**